### ISSUE TITLE
example?

### ISSUE THREAD

**Issue by koenverburg** [Folkloreatelier/laravel-hypernova#1]
Can you make an example. The [Hypernova-ruby](https://github.com/airbnb/hypernova-ruby) uses this in there templating engine.

```ruby
<%=
  render_react_component(
    'MyComponent.js',
    :name => 'Person',
    :color => 'Blue',
    :shape => 'Triangle'
  )
%>
```

Can we write something like this to render a component which is wrapped with hypernova-react?

```php

{{
  render_react_component( 'MyComponent.js', [
    'name' => 'Person', 'props' => 'value' 
  ])
}}
```

**Comment by koenverburg** [comment]
Can you give an update on this?

**Comment by dmongeau** [comment]
Yes there is a blade directive called hypernova that you can use like in your example:

```php
@hypernova( 'MyComponent.js', [
    'name' => 'Person', 'props' => 'value' 
])

```

**Comment by eightyfive** [comment]
Directive only works when Middleware is activated:
```php
// app/Http/Kernel.php
protected $middleware = [
    ...
    \Folklore\Hypernova\HypernovaMiddleware::class,
];
```

@dmongeau correct me if I'm wrong:
Directive only outputs placeholders. They are replaced in Middleware by corresponding Hypernova job result (HTML string).

Please note that there seems to be a [bug in the Middleware](https://github.com/Folkloreatelier/laravel-hypernova/issues/2).

Finally, the following blog post helped me to get started:
https://blog.larah.me/2017/04/10/react-server-side-rendering/

**Comment by dmongeau** [comment]
Yes you are right, you need the middleware, because by outputting only a placeholder, if the rendering server doesn't work, the client code render the UI instead. But I really have a job to make this package "more public" with doc didn't expect, it will be used in this state. But I'm open to any PR improving it

**Comment by aftabnaveed** [comment]
@dmongeau Does each call to @hypernova directive addJob  to the batch? And all components are then sent to the server at once?

**Comment by aftabnaveed** [comment]
I can answer my question now and yes it does!
